pkg/setting/settings.go

```diff
@@ -0,0 +1,140 @@
+package setting
```

Comment:
Just a nit, but to avoid confusion with pkg/setting/setting.go, could this file be named e.g. provider.go instead?